cmd/plugin/builder/template/plugintemplates/main.go.tmpl

```diff
@@ -11,6 +12,7 @@ import (
 var descriptor = plugin.PluginDescriptor{
 	Name:        "{{ .PluginName | ToLower }}",
 	Description: "{{ .Description | ToLower }}",
+	Target:      types.TargetUnknown,
```

Comment:
Not sure if you intended this or not

when running `make plugin-build-local`:

```
[x] : invalid PluginDescriptor specified: plugin "hello": target is not valid
exit status 1
make: *** [plugin-build-darwin-arm64] Error 1
```

If we want the developer to consciously fix this, we need to add a comment at the minimum. 
And even with a comment, I wonder if having something that compiles out of the gate is better or not.

Reply:
If we end up with this behavior, we may consider being even more in-your-face about what to fix:
like changing
```
Target:      types.TargetUnknown, // set the Target of the plugin. Possible values are TargetGlobal, TargetK8s, TargetTMC
```
to
```
Target:      "FIXME<<<", // set the Target of the plugin to types.(one of {TargetGlobal,TargetK8s,TargetTMC})
```
because types.TargetUnknown looks suspiciously like a value that could be allowed.

I am okay with the change changes for now,  but I think as a followup enhancement, we might want consider asking the user for the target value, and applying the correct one so the project builds correctly.

Reply:
Sure. That sounds reasonable as well. Updated.

Reply:
Documentation update PR: https://github.com/vmware-tanzu/tanzu-cli/pull/160